src/Adapter/Carrier/CommandHandler/ToggleCarrierIsFreeHandler.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Carrier\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Carrier\AbstractCarrierHandler;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\Command\ToggleCarrierIsFreeCommand;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\CommandHandler\ToggleCarrierIsFreeHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\Exception\CannotToggleCarrierIsFreeStatusException;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\Exception\CarrierException;
+use PrestaShopException;
+
+class ToggleCarrierIsFreeHandler extends AbstractCarrierHandler implements ToggleCarrierIsFreeHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(ToggleCarrierIsFreeCommand $command)
+    {
+        $carrier = $this->getCarrier($command->getCarrierId());
+
+        try {
+            $carrier->setFieldsToUpdate(['is_free' => true]);
+            $carrier->is_free = !(bool) $carrier->is_free;
+
+            if (false === $carrier->update()) {
+                throw new CannotToggleCarrierIsFreeStatusException(sprintf('Unable to toggle is-free status of carrier with id "%d"', $command->getCarrierId()->getValue()));
```

Comment:
I don't understand what is "is-free status" could you provide me a screenshot to see where it appears please?

Reply:
![Screenshot 2020-10-26 at 16 29 47](https://user-images.githubusercontent.com/45097959/97185315-9368de00-17a8-11eb-9d8d-806b09d695f1.png)

This is free shipping, just with other name. Pic from not migrated page.

Reply:
I had never heard this word before... Is it supposed to replace "Free shipping"? Why don't we keep "Free shipping" here?

Reply:
This stay Free Shipping in UI, It's is-free only in code

Reply:
I am sorry but I still don't understand why it appears in the wording then... ?

![image](https://user-images.githubusercontent.com/70583503/97304520-2282fe00-185c-11eb-9d65-e8b9b1f2f2fb.png)

Reply:
HI @ks129! I am just waiting for your answer about the is free status before I can validate the wording. ☺️ I don't understand why the word "is-free" appears in this sentence " An error occurred while updating the is-free status." since you told me it should only appear in the code... 🤔 Thank you in advance for your answer.

Reply:
Sorry, I forget fixing it. Now in error msg this is renamed to free shipping @Julievrz